### PR TITLE
convert `assignWithDepth` to TS

### PR DESCRIPTION
**Pull request by Yokozuna59**
## :bookmark_tabs: Summary

Brief description about the content of your PR.

Resolves #<your issue id here>

## :straight_ruler: Design Decisions

Describe the way your implementation works or what design decisions you made if applicable.

### :clipboard: Tasks

Make sure you

- [ ] :book: have read the [contribution guidelines](https://github.com/mermaid-js/mermaid/blob/develop/CONTRIBUTING.md)
- [ ] :computer: have added necessary unit/e2e tests.
- [ ] :notebook: have added documentation. Make sure [`MERMAID_RELEASE_VERSION`](https://github.com/mermaid-js/mermaid/blob/develop/packages/mermaid/src/docs/community/development.md#3-update-documentation) is used for all new features.
- [ ] :bookmark: targeted `develop` branch

### CODE REVIEW

File: packages/mermaid/src/assignWithDepth.ts [hunk 1]
```diff
@@ -1,32 +1,36 @@
...
+ * * clobber: should dissimilar types clobber
  */
-const assignWithDepth = function (dst, src, config) {
-  const { depth, clobber } = Object.assign({ depth: 2, clobber: false }, config);
+const assignWithDepth = (
+  dst: any,
+  src: any,
+  config: { depth: number; clobber: boolean } = { depth: 2, clobber: false }
+): any => {
+  const { depth, clobber } = config;
```
Comment (sidharthv96) [inline packages/mermaid/src/assignWithDepth.ts:33]: The Object.assign cannot be removed. 

If you call `assignWithDepth(..,.., { depth: 2 })`, clobber will be undefined now. 


This is something else you can do. 
```suggestion
  {depth = 2, clobber = false}: { depth: number; clobber: boolean } = {}
): any => {
  const config = { depth, clobber };
```

Comment (Yokozuna59) [reply packages/mermaid/src/assignWithDepth.ts:33]: I'm not sure I'm following; how would it be `undefined`? It has a default value, so even if we didn't pass any of them, it would use those values.
Can't we just re-add `Object.assign`?

Comment (sidharthv96) [reply packages/mermaid/src/assignWithDepth.ts:33]: > I'm not sure I'm following; how would it be undefined? It has a default value, so even if we didn't pass any of them, it would use those values.

Yes, it has a default value, `{ depth: 2, clobber: false }`. 
But if you pass a partial value when calling, like `assignWithDepth(..,.., { depth: 2 })`, that default value will not be used, so, the config will have the value `{ depth: 2 }`. Which makes `clobber` undefined. 

In short, the default value is not "merged" with the passed value, but it's used only if there is no passed value. 

We have to use the format in my suggestion to put default values into individual properties, essentially emitating merge.

Comment (sidharthv96) [reply packages/mermaid/src/assignWithDepth.ts:33]: Yes, you can re-add the assign too.

Comment (Yokozuna59) [reply packages/mermaid/src/assignWithDepth.ts]: I see, new to JS/TS world so I don't know much :)

But the suggested code isn't a valid code, I remove the `= {}` part. f02dd0a5fc269910b9da3822bd76264206210ba8

Comment (sidharthv96) [reply packages/mermaid/src/assignWithDepth.ts]: I'm sorry, but I forgot the `?` in the type. This should be valid. 
It is not necessary to put the values in two places. 

```ts
  {depth = 2, clobber = false}: { depth?: number; clobber?: boolean } = {}
```

Comment (sidharthv96) [reply packages/mermaid/src/assignWithDepth.ts]: > I see, new to JS/TS world so I don't know much :)

Well, I just found out about `structuredClone` and [`queueMicrotask`](https://developer.mozilla.org/en-US/docs/Web/API/queueMicrotask) last week, so we're all learning :)

File: packages/mermaid/src/assignWithDepth.ts [hunk 2]
```diff
@@ -1,32 +1,36 @@
...
+ * * clobber: should dissimilar types clobber
  */
-const assignWithDepth = function (dst, src, config) {
-  const { depth, clobber } = Object.assign({ depth: 2, clobber: false }, config);
+const assignWithDepth = (
+  dst: any,
+  src: any,
+  { depth = 2, clobber = false }: { depth: number; clobber: boolean } = { depth: 2, clobber: false }
+): any => {
+  const config = Object.assign({ depth, clobber });
```
Comment (sidharthv96) [inline packages/mermaid/src/assignWithDepth.ts:33]: Btw, you don't need both the initializer and object.assign. Either one is fine.
